lib/octokit/client/source_import.rb

```diff
@@ -0,0 +1,90 @@
+module Octokit
+  class Client
+
+    # Methods for the Source Import API
+    #
+    # @see https://developer.github.com/v3/repos/source_imports
+    module SourceImport
+
+      # Start a source import to a GitHub repository using GitHub Importer.
+      #
+      # @param repo [Integer, String, Hash, Repository] A GitHub repository.
+      # @param vcs [String] The originating VCS type. Can be one of "subversion", "git", "mercurial", or "tfvc".
+      # @param vcs_url [String] The URL of the originating repository.
+      # @param options [Hash]
+      # @option options [String] :vcs_username If authentication is required, the username to provide to vcs_url.
+      # @option options [String] :vcs_password If authentication is required, the password to provide to vcs_url.
+      # @option options [String] :tfvc_project For a tfvc import, the name of the project that is being imported.
+      # @return [Sawyer::Resource] Hash representing the repository import
+      # @see https://developer.github.com/v3/migration/source_imports/#start-an-import
+      #
+      # @example
+      #   @client.import("octokit/octokit.rb", "subversion", "http://svn.mycompany.com/svn/myproject" {
+      #    :vcs_username" => "octocat",
+      #    :vcs_password  => "secret"
+      #   })
+      def start_source_import(repo, vcs, vcs_url, options = {})
+        options = ensure_api_media_type(:source_imports, options.merge(:vcs => vcs, :vcs_url => vcs_url))
+        put "#{Repository.path repo}/import", options
+      end
+
+      # View the progress of an import.
+      #
+      # @param repo [Integer, String, Hash, Repository] A GitHub repository.
+      # @return [Sawyer::Resource] Hash representing the progress of the import
+      # @see https://developer.github.com/v3/migration/source_imports/#get-import-progress
+      #
+      # @example
+      #   @client.source_import_progress("octokit/octokit.rb")
+      def source_import_progress(repo, options = {})
+        options = ensure_api_media_type(:source_imports, options)
+        get "#{Repository.path repo}/import", options
+      end
+
+      # List source import commit authors
+      #
+      # @param repo [Integer, String, Hash, Repository] A GitHub repository.
+      # @return [Array<Sawyer::Resource>] Array of hashes representing commit_authors.
```

Comment:
It looks like there's a [`since` option](https://developer.github.com/v3/migration/source_imports/#parameters-1) in the hash, too.